packages/nimble-components/src/rich-text/models/markdown-parser.ts

```diff
@@ -46,10 +51,44 @@ export class RichTextMarkdownParser {
             'autolink'
         ]);
 
+        supportedTokenizerRules.validateLink = href => /^https?:\/\//i.test(href);
+
+        // In order to display encoded characters, non-ASCII characters, emojis, and other special characters in their original form,
+        // we bypass the default normalization of link text in markdown-it. This is done because we support only "AutoLink" feature in CommonMark flavor.
+        // "normalizeLinkText" method reference in markdown-it: https://github.com/markdown-it/markdown-it/blob/2b6cac25823af011ff3bc7628bc9b06e483c5a08/lib/index.js#L67C1-L86C2
+        supportedTokenizerRules.normalizeLinkText = url => url;
```

Comment:
So what's going to happen when we support more than the autolink feature? Won't this impact both autolink and normal link cases?

Reply:
This code needs an update when we enable normal link support. I have mentioned this along with other updates here to track the work: https://github.com/ni/nimble/issues/1527